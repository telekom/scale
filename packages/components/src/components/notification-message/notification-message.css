/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

:host {
  --width: 100%;
  --radius: var(--telekom-radius-standard);
  --border: var(--telekom-line-weight-standard) solid transparent;

  --background-error: var(--telekom-color-functional-danger-subtle);
  --background-warning: var(--telekom-color-functional-warning-subtle);
  --background-informational: var(
    --telekom-color-functional-informational-subtle
  );
  --background-success: var(--telekom-color-functional-success-subtle);
}

.notification-message {
  border: var(--border);
  border-radius: var(--radius);
  width: var(--width);
  position: relative;
}

.notification-message--variant-error {
  background: var(--background-error);
}
.notification-message--variant-warning {
  background-color: var(--background-warning);
}
.notification-message--variant-informational {
  background-color: var(--background-informational);
}
.notification-message--variant-success {
  background-color: var(--background-success);
}

.notification-message__heading {
  margin: 14px 48px 14px 48px;
  font-weight: var(--telekom-typography-font-weight-bold);
  line-height: var(--telekom-typography-line-spacing-standard);
}

.notification-message__text {
  margin: 8px 48px 15px 48px;
}

::slotted(*) {
  font-weight: var(--telekom-typography-font-weight-regular);
}

.notification-message__icon-close {
  position: absolute;
  top: 8px;
  right: 13.5px;
  color: var(--telekom-color-text-and-icon-standard);
}

.notification-message__icon-close svg {
  height: 19px;
  width: 19px;
  padding: 6.5px;
  border-radius: 20%;
}

.notification-message__icon-close:hover svg {
  background-color: hsl(0, 0%, 100%);
  color: var(--telekom-color-text-and-icon-primary-hovered);
}

.notification-message__icon-success {
  position: absolute;
  top: 12.5px;
  left: 17px;
  height: 20px;
  width: 20px;
  color: var(--telekom-color-text-and-icon-functional-success);
}

.notification-message__icon-error {
  position: absolute;
  top: 12px;
  left: 17px;
  height: 20px;
  width: 20px;
  color: var(--telekom-color-text-and-icon-functional-danger);
}

.notification-message__icon-information {
  position: absolute;
  top: 12px;
  left: 17px;
  height: 20px;
  width: 20px;
  color: var(--telekom-color-text-and-icon-functional-informational);
}

@media screen and (forced-colors: active), (-ms-high-contrast: active) {
  .notification-message__icon-close {
    color: hsl(0, 0%, 100%);
  }
}
