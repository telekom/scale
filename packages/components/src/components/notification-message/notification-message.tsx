/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

import { Component, h, Host, Prop, Element, Method } from '@stencil/core';
import classNames from 'classnames';
import statusNote from '../../utils/status-note';

@Component({
  tag: 'scale-notification-message',
  styleUrl: 'notification-message.css',
  shadow: true,
})
export class NotificationMessage {
  @Element() hostElement: HTMLElement;

  @Prop() variant?: 'informational' | 'success' | 'warning' | 'error' =
    'informational';
  @Prop() dismissible?: boolean = false;
  @Prop({ reflect: true }) opened: boolean;
  @Prop() autoHide?: boolean = false;
  @Prop() autoHideDuration?: number = 3000;

  hasSlotText: boolean;

  componentDidLoad() {
    this.hasSlotText = !!this.hostElement.querySelector("p[slot='text']");
  }

  componentDidRender() {
    if (this.autoHide === true) {
      setTimeout(this.close, this.autoHideDuration);
    }
  }

  connectedCallback() {
    statusNote({ source: this.hostElement, type: 'warn' });
  }

  @Method()
  async open() {
    this.opened = true;
  }

  handleIcons() {
    if (this.variant) {
      switch (this.variant) {
        case 'success':
          return (
            <scale-icon-alert-success
              class="notification-message__icon-success"
              accessibility-title="success"
              color="#187431"
            />
          );
        case 'informational':
          return (
            <scale-icon-alert-information
              class="notification-message__icon-information"
              accessibility-title="information"
            />
          );
        case 'error':
          return (
            <scale-icon-alert-error
              class="notification-message__icon-error"
              accessibility-title="error"
            />
          );
        case 'warning':
          return (
            <scale-icon-alert-error
              class="notification-message__icon-information"
              color="#AE461C"
              accessibility-title="information"
            />
          );
      }
    }
    return;
  }

  close = () => {
    this.opened = false;
  };

  render() {
    if (!this.opened) {
      return null;
    }

    return (
      <Host>
        <div
          part={this.getBasePartMap()}
          class={this.getCssClassMap()}
          tabindex="0"
        >
          <div part="container" class="notification-message__container">
            {this.handleIcons()}
            <div part="heading" class="notification-message__heading">
              <slot>&emsp;</slot>

              {this.dismissible && (
                <scale-icon-action-circle-close
                  tabindex="0"
                  class="notification-message__icon-close"
                  onClick={() => {
                    this.close();
                  }}
                  onKeyDown={(e) => {
                    if (e.key === 'Enter') {
                      this.close();
                    }
                  }}
                  accessibility-title="close"
                />
              )}
            </div>
            {this.hasSlotText && (
              <div part="text" class="notification-message__text">
                <slot name="text" />
              </div>
            )}
          </div>
        </div>
      </Host>
    );
  }

  getBasePartMap() {
    return this.getCssOrBasePartMap('basePart');
  }

  getCssClassMap() {
    return this.getCssOrBasePartMap('css');
  }

  getCssOrBasePartMap(mode: 'basePart' | 'css') {
    const name = 'notification-message';
    const prefix = mode === 'basePart' ? '' : `${name}--`;

    return classNames(name, this.variant && `${prefix}variant-${this.variant}`);
  }
}
