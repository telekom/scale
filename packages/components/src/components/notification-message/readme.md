# scale-notification-message

<!-- Auto Generated Below -->


## Properties

| Property           | Attribute            | Description | Type                                                   | Default           |
| ------------------ | -------------------- | ----------- | ------------------------------------------------------ | ----------------- |
| `autoHide`         | `auto-hide`          |             | `boolean`                                              | `false`           |
| `autoHideDuration` | `auto-hide-duration` |             | `number`                                               | `3000`            |
| `dismissible`      | `dismissible`        |             | `boolean`                                              | `false`           |
| `opened`           | `opened`             |             | `boolean`                                              | `undefined`       |
| `variant`          | `variant`            |             | `"error" \| "informational" \| "success" \| "warning"` | `'informational'` |


## Events

| Event         | Description                                            | Type                |
| ------------- | ------------------------------------------------------ | ------------------- |
| `scale-close` | Fires when the notification message has been dismissed | `CustomEvent<void>` |


## Methods

### `open() => Promise<void>`



#### Returns

Type: `Promise<void>`




## Shadow Parts

| Part                   | Description |
| ---------------------- | ----------- |
| `"button-dismissable"` |             |
| `"container"`          |             |
| `"heading"`            |             |
| `"text"`               |             |


## Dependencies

### Depends on

- [scale-icon-alert-success](../icons/alert-success)
- [scale-icon-alert-information](../icons/alert-information)
- [scale-icon-alert-error](../icons/alert-error)
- [scale-icon-alert-warning](../icons/alert-warning)
- [scale-icon-action-circle-close](../icons/action-circle-close)

### Graph
```mermaid
graph TD;
  scale-notification-message --> scale-icon-alert-success
  scale-notification-message --> scale-icon-alert-information
  scale-notification-message --> scale-icon-alert-error
  scale-notification-message --> scale-icon-alert-warning
  scale-notification-message --> scale-icon-action-circle-close
  style scale-notification-message fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
