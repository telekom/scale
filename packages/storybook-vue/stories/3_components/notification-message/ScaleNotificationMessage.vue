<template>
  <div>
    <slot name="text" />
  </div>
</template>

<script>
export default {
  props: {
    variant: {
      type: 'error' | 'warning' | 'success' | 'informational',
      default: 'informational',
    },
    dismissible: { type: Boolean, default: false },
    opened: { type: Boolean },
    timeout: { type: Boolean | Number, default: false },
  },
};
</script>
