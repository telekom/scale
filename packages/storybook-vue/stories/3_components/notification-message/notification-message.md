<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Notification Message</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>

The standard Notification Message component can be used to output a notification message. In addition, a short explanatory text can be added to it.

## Beta components

This component is still in the beta phase. When testing it, keep in mind that it may not have gone through all quality control measures, and it may not yet have WCAG accessibility certification. There may be changes to this component in the future.
