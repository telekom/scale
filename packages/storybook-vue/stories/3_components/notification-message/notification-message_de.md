<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Notification Message</h1>
    <img src="assets/beta.png" alt="Beta Component" />
</div>
Die standardmäßige Notification Message Komponente kann zum ausgeben von einer Mitteilung/ Meldung genutzt werden. Ergänzend kann dieser auch ein kurzer erklärender Text hinzugefügt werden.

## Beta-Komponente

Diese Komponente befindet sich noch im Beta-Stadium. Wenn du sie testest, bedenke, dass sie möglicherweise noch nicht alle Qualitätskontrollmaßnahmen durchlaufen hat und noch keine WCAG-Zertifizierung zur Barrierefreiheit vorliegt. In Zukunft kann es zu Änderungen an dieser Komponente kommen.
