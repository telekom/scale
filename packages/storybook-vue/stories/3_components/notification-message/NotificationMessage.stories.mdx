import {
  Meta,
  Story,
  ArgsTable,
  Canvas,
  Description,
} from '@storybook/addon-docs/blocks';
import ScaleNotificationMessage from './ScaleNotificationMessage.vue';

<Meta
  title="Beta Components/Notification Message"
  component={ScaleNotificationMessage}
  argTypes={{
    variant: {
      defaultValue: 'informational',
      options: ['error', 'warning', 'success', 'informational'],
      control: {
        type: 'select',
      },
    },
    hasClose: {
      defaultValue: false,
      options: [true, false],
      control: {
        type: 'select',
      },
    },
    opened: {
      options: [true, false],
      control: {
        type: 'select',
      },
    },
    timout: {
      defaultValue: false,
      options: [false, true, Number],
      control: {
        type: 'select',
      },
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  components: { ScaleNotificationMessage },
  props: {
    ...ScaleNotificationMessage.props,
  },
  template: `
    <scale-notification-message   
     :variant="variant"
     :hasClose="hasClose"
     :timeout="timeout"
     :opened="opened"
    >
    </scale-notification-message>
  `,
});

export const TemplateSlot = (args, { argTypes }) => ({
  components: { ScaleNotificationMessage },
  props: {
    ...ScaleNotificationMessage.props,
  },
  template: `
    <scale-notification-message   
     :variant="variant"
     :hasClose="hasClose"
     :timeout="timeout"
     :opened="opened"
    >
    <template v-slot:header>
     <p slot="header">This is my title</p>
    </template>
    <template v-slot:text>
     <p slot="text">This is my message to you</p>
    </template>
  `,
});

## Standard

### All default values

<Canvas withSource="none">
  <Story
    name="Standard"
    args={{
      opened: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />

```html
<scale-notification-message opened> </scale-notification-message>
```

## Scoped CSS variables

```css
:host {
  /* NotificationMessage styling */
  --notification-message-border-radius: 8px;
  --notification-message-border: 1px solid var(--scl-color-white);
  --notification-message-width: 100%;
  --notification-message-font-header: var(--scl-fonts-family-sans);
  --notification-message-font-size-header: 16px;
  --notification-message-font-text: var(--scl-font-weight-regular);
  --notification-message-font-size-text: 16px;
  --variant-error: hsla(353.79999999999995, 100%, 88.6%, 0.35);
  --variant-warning: hsla(33.5, 100%, 75.1%, 0.3);
  --variant-informational: hsla(190.20000000000005, 100%, 92%, 0.75);
  --variant-success: hsl(98.80000000000001, 49.7%, 61%, 0.2);
}
```

## Variants

### Warning

<Canvas withSource="none">
  <Story
    name="Warning"
    args={{
      opened: true,
      variant: 'warning',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="warning" opened>
</scale-notification-message>
```

### Error

<Canvas withSource="none">
  <Story
    name="Error"
    args={{
      opened: true,
      variant: 'error',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="error" opened>
</scale-notification-message>
```

### Success

<Canvas withSource="none">
  <Story
    name="Success"
    args={{
      opened: true,
      variant: 'success',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="success" opened>
</scale-notification-message>
```

### Informational

<Canvas withSource="none">
  <Story
    name="Informational"
    args={{
      opened: true,
      variant: 'informational',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="informational" opened>
</scale-notification-message>
```

## Close Button

<Canvas withSource="none">
  <Story
    name="Close Button"
    args={{
      opened: true,
      hasClose: true,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message has-close opened> </scale-notification-message>
```

## Timeout

### Set to close in 10 seconds. Default - when true: 3 seconds

<Canvas withSource="none">
  <Story
    name="Timeout"
    args={{
      opened: true,
      timeout: 10000,
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message timeout="10000" opened>
</scale-notification-message>
```

## Titel & Message

<Canvas withSource="none">
  <Story
    name="Titel-Message"
    args={{
      opened: true,
      hasClose: true,
    }}
  >
    {TemplateSlot.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message opened>
  <p slot="header">This is my title</p>
  <p slot="text">This is my message to you.</p>
</scale-notification-message>
```
