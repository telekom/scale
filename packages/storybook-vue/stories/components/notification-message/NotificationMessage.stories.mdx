import {
  Meta,
  Story,
  ArgsTable,
  Canvas,
  Description,
} from '@storybook/addon-docs';
import ScaleNotificationMessage from './ScaleNotificationMessage.vue';

<Meta
  title="Beta Components/Notification Message"
  component={ScaleNotificationMessage}
  argTypes={{
    variant: {
      defaultValue: 'informational',
      options: ['error', 'warning', 'success', 'informational'],
      control: {
        type: 'select',
      },
    },
  }}
/>

export const Template = (args, { argTypes }) => ({
  props: {
    ...ScaleNotificationMessage.props,
    heading: String,
    text: String,
  },
  template: `
    <scale-notification-message   
      :variant="variant"
      :dismissible="dismissible"
      :auto-hide-duration="autoHideDuration"
      :auto-hide="autoHide"
      :opened="opened"
    >
      {{ heading }} 
      <p v-if="text">{{ text }}</p>
    </scale-notification-message>
  `,
});

<div
  style={{
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
  }}
>
  <h1>Notification Message</h1>
  <img src="assets/beta.png" alt="Beta Component" />
</div>

## Standard

<Canvas withSource="none">
  <Story
    name="Standard"
    args={{
      opened: true,
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

<ArgsTable story="Standard" />

```html
<scale-notification-message opened
  >Notification text goes here</scale-notification-message
>
```

## Scoped CSS variables

```css
:host {
  --width: 100%;
  --radius: var(--telekom-radius-standard);
  --border: var(--telekom-line-weight-standard) solid transparent;

  --background-error: var(--telekom-color-functional-danger-subtle);
  --background-warning: var(--telekom-color-functional-warning-subtle);
  --background-informational: var(
    --telekom-color-functional-informational-subtle
  );
  --background-success: var(--telekom-color-functional-success-subtle);
}
```

## Variants

### Warning

<Canvas withSource="none">
  <Story
    name="Warning"
    args={{
      opened: true,
      variant: 'warning',
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="warning" opened>
  Notification text goes here
</scale-notification-message>
```

### Error

<Canvas withSource="none">
  <Story
    name="Error"
    args={{
      opened: true,
      variant: 'error',
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="error" opened>
  Notification text goes here
</scale-notification-message>
```

### Success

<Canvas withSource="none">
  <Story
    name="Success"
    args={{
      opened: true,
      variant: 'success',
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="success" opened>
  Notification text goes here
</scale-notification-message>
```

### Informational

<Canvas withSource="none">
  <Story
    name="Informational"
    args={{
      opened: true,
      variant: 'informational',
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message variant="informational" opened>
  Notification text goes here
</scale-notification-message>
```

## Dismissible

<Canvas withSource="none">
  <Story
    name="Dismissible"
    args={{
      opened: true,
      dismissible: true,
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message dismissible opened>
  Notification text goes here
</scale-notification-message>
```

## Timeout

<Canvas withSource="none">
  <Story
    name="Timeout"
    args={{
      opened: true,
      autoHideDuration: 4000,
      autoHide: true,
      heading: 'Notification text goes here',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message auto-hide-duration="4000" auto-hide opened>
  Notification text goes here
</scale-notification-message>
```

## With Text

<Canvas withSource="none">
  <Story
    name="With Text"
    args={{
      opened: true,
      heading: 'Notification text goes here',
      text: 'Here you can add some more text',
    }}
  >
    {Template.bind({})}
  </Story>
</Canvas>

```html
<scale-notification-message opened>
  Notification text goes here.
  <p slot="text">Here you can add some more text.</p>
</scale-notification-message>
```
