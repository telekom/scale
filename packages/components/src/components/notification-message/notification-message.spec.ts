import { newSpecPage } from '@stencil/core/testing';
// import { remove } from 'lodash';
import { NotificationMessage } from './notification-message';

describe('NotificationMessage', () => {
  let element;

  beforeEach(async () => {
    element = new NotificationMessage();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [NotificationMessage],
      html: `<scale-notification-message opened="true"></scale-notification-message>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.variant = 'warning';
    expect(element.getCssClassMap()).toContain('variant-warning');
  });

  it('should test the close() fucntion', () => {
    expect(element.opened).toBe(undefined);
    element.close();
    expect(element.opened).toBe(false);
  });

  it('handle variant prop', async () => {
    const page = await newSpecPage({
      components: [NotificationMessage],
      html: `<scale-notification-message opened="true">Label</scale-notification-message>`,
    });
    page.root.variant = 'error';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    page.root.variant = 'warning';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    page.root.variant = 'informational';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
    page.root.variant = 'success';
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('should emit onClick and set opened to false', async () => {
    const page = await newSpecPage({
      components: [NotificationMessage],
      html: `<scale-notification-message opened="true" dismissible="true">Label</scale-notification-message>`,
    });
    const icon = page.root.shadowRoot.querySelector(
      '.notification-message__icon-close'
    );
    await page.waitForChanges();
    icon.dispatchEvent(new Event('click'));
    await page.waitForChanges();
    expect(page.rootInstance.opened).toBe(false);
  });
});
