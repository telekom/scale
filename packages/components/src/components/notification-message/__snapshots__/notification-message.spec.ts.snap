// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationMessage handle variant prop 1`] = `
<scale-notification-message opened="">
  <mock:shadow-root>
    <div class="notification-message notification-message--variant-error" part="notification-message variant-error" tabindex="0">
      <div class="notification-message__container" part="container">
        <scale-icon-alert-warning accessibility-title="error" class="notification-message__icon-error"></scale-icon-alert-warning>
        <div class="notification-message__heading" part="heading">
          <slot></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-message>
`;

exports[`NotificationMessage handle variant prop 2`] = `
<scale-notification-message opened="">
  <mock:shadow-root>
    <div class="notification-message notification-message--variant-warning" part="notification-message variant-warning" tabindex="0">
      <div class="notification-message__container" part="container">
        <scale-icon-alert-information accessibility-title="information" class="notification-message__icon-information" color="#AE461C"></scale-icon-alert-information>
        <div class="notification-message__heading" part="heading">
          <slot></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-message>
`;

exports[`NotificationMessage handle variant prop 3`] = `
<scale-notification-message opened="">
  <mock:shadow-root>
    <div class="notification-message notification-message--variant-informational" part="notification-message variant-informational" tabindex="0">
      <div class="notification-message__container" part="container">
        <scale-icon-alert-information accessibility-title="information" class="notification-message__icon-information"></scale-icon-alert-information>
        <div class="notification-message__heading" part="heading">
          <slot></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-message>
`;

exports[`NotificationMessage handle variant prop 4`] = `
<scale-notification-message opened="">
  <mock:shadow-root>
    <div class="notification-message notification-message--variant-success" part="notification-message variant-success" tabindex="0">
      <div class="notification-message__container" part="container">
        <scale-notification-message-svg accessibility-title="success" class="notification-message__icon-success"></scale-notification-message-svg>
        <div class="notification-message__heading" part="heading">
          <slot></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  Label
</scale-notification-message>
`;

exports[`NotificationMessage should match snapshot 1`] = `
<scale-notification-message opened="">
  <mock:shadow-root>
    <div class="notification-message notification-message--variant-informational" part="notification-message variant-informational" tabindex="0">
      <div class="notification-message__container" part="container">
        <scale-icon-alert-information accessibility-title="information" class="notification-message__icon-information"></scale-icon-alert-information>
        <div class="notification-message__heading" part="heading">
          <slot></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</scale-notification-message>
`;
